src/Editor/Util/RayIntersection.cpp

```diff
@@ -0,0 +1,65 @@
+#include "RayIntersection.hpp"
+
+RayIntersection::RayIntersection() {
+    
+}
+
+RayIntersection::~RayIntersection() {
+    
+}
+
+bool RayIntersection::RayOBBIntersect(glm::vec3 rayOrigin, glm::vec3 rayDirection, Video::AxisAlignedBoundingBox meshData, glm::mat4 modelMatrix, float &outputDistance) {
+
+    float tMin = -INFINITY;
+    float tMax = INFINITY;
+  
+    glm::vec3 worldPos = glm::vec3(modelMatrix[3].x, modelMatrix[3].y, modelMatrix[3].z);
+    glm::vec3 delta = worldPos - rayOrigin;
+
+    glm::vec3 xyzAxis[3];
+    float minValue[3] = { meshData.minVertex.x, meshData.minVertex.y, meshData.minVertex.z };
+    float maxValue[3] = { meshData.maxVertex.x, meshData.maxVertex.y, meshData.maxVertex.z };
+
+    for (int i = 0; i < 3; i++) {
+        xyzAxis[i] = glm::vec3(modelMatrix[i].x, modelMatrix[i].y, modelMatrix[i].z);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'xyzAxis' is assigned a value that is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)